File: 1631-number-of-sub-arrays-with-odd-sum/number-of-sub-arrays-with-odd-sum.py
class Solution:
    def numOfSubarrays(self, arr: List[int]) -> int:
        cnt = [1, 0]
        pref, ans = 0, 0
        for i in arr:
            pref += i
            ans += cnt[pref & 1 ^ 1]
            cnt[pref & 1] += 1 
        return ans % 1_000_000_007

File: 1631-number-of-sub-arrays-with-odd-sum/number-of-sub-arrays-with-odd-sum.java
class Solution {
    public int numOfSubarrays(int[] arr) {
        long[] cnt = new long[2] ;
        int ans = 0 ;
        int pref = 0;
        cnt[0] = 1;
        for(int i : arr){
            pref += i ;
            ans += cnt[(pref & 1) ^ 1];
            ans %= 1000000007 ;
            cnt[pref & 1] ++ ;
        }
        return ans ;
    }
}